common/src/java/org/apache/hadoop/hive/common/LogUtils.java

```diff
@@ -245,16 +245,14 @@ public static void unregisterLoggingContext() {
   public static String getLogFilePath() {
     String logFilePath = null;
     org.apache.logging.log4j.Logger rootLogger = LogManager.getRootLogger();
-    if (rootLogger instanceof org.apache.logging.log4j.core.Logger) {
-      org.apache.logging.log4j.core.Logger coreLogger =
-          (org.apache.logging.log4j.core.Logger)rootLogger;
+    if (rootLogger instanceof org.apache.logging.log4j.core.Logger coreLogger) {
       for (Appender appender : coreLogger.getAppenders().values()) {
-        if (appender instanceof FileAppender) {
-          logFilePath = ((FileAppender) appender).getFileName();
-        } else if (appender instanceof RollingFileAppender) {
-          logFilePath = ((RollingFileAppender) appender).getFileName();
-        } else if (appender instanceof RollingRandomAccessFileAppender) {
-          logFilePath = ((RollingRandomAccessFileAppender) appender).getFileName();
+        if (appender instanceof FileAppender fileAppender) {
```

Comment:
could extract the snippet into 'getLogFilePath(Appender appender)' and use switch here?
````
switch (appender) {
  case FileAppender fa -> fa.getFileName()
  ..
}
````
do we need break the looks when condition match?